crates/lib3h/src/engine/real_engine.rs

```
@@ -101,48 +99,88 @@ impl<'engine> RealEngine<'engine> {
-        let network_transport = TransportWrapper::new(TransportMemory::new());
-        // Bind & create DhtConfig
-        let binding = network_transport
-            .as_mut()
-            .bind(&config.bind_url)
-            .expect("TransportMemory.bind() failed. bind-url might not be unique?");
-        let dht_config =
-            DhtConfig::with_real_engine_config(format!("{}_tId", name).as_str(), &binding, &config);
+        let mut memory_transport = GhostTransportMemory::new();
+        let mut memory_network_endpoint = Detach::new(
+            memory_transport
+                .take_parent_endpoint()
+                .expect("exists")
+                .as_context_endpoint_builder()
+                .request_id_prefix("tmem_to_child_")
+                .build::<GatewayUserData, Lib3hTrace>(),
+        );
+
+        // Bind & create this_net_peer
+        let mut gateway_ud = GatewayUserData::new();
+        let _res = memory_network_endpoint.request(
+            Lib3hTrace,
+            transport::protocol::RequestToChild::Bind {
+                spec: config.bind_url.clone(),
+            },
+            Box::new(|mut ud, response| {
+                let response = {
+                    match response {
+                        GhostCallbackData::Timeout => panic!("timeout"),
+                        GhostCallbackData::Response(response) => match response {
+                            Err(e) => panic!("{:?}", e),
+                            Ok(response) => response,
+                        },
+                    }
+                };
+                if let transport::protocol::RequestToChildResponse::Bind(bind_data) = response {
+                    ud.binding = bind_data.bound_url;
+                } else {
+                    panic!("bad response to bind: {:?}", response);
+                }
+                Ok(())
+            }),
+        );
+        let _ = memory_transport.process();
```

We should `?` this, or at the very least `unwrap()` rather than ignoring errors